tests/functional/cross_db_utils/fixture_escape_single_quotes.py

```diff
@@ -0,0 +1,26 @@
+
+# escape_single_quotes
```

Comment:
## The alternative

However, the single test case with the `escape_character` definition was really interesting. I wonder if it would be worth adding the `escape_character` definition within each adapter itself?

## An idea
Create a collection of low-level definitions within each adapter:
- `literal_quote` - the symbol used to demarcate a (string) literal value within the database.
    - ANSI standard is single quote.
    - We'd then use this within the default implementation of the `string_literal` macro.
- `standard_escape_character` - The symbol used to interpret a character literally rather than interpret it.
    - The ANSI standard is the backslash character.
- `invalid_for_standard_escape` - List of symbols that can't be escaped by the `standard_escape_character`.
    -  The ANSI standard includes both single and double quotes.
    - We'd default this to the `literal_quote` value (and maybe double quote too).
- `special_escape_character` - The symbol used to escape `invalid_for_standard_escape` characters.
    - The ANSI standard is a single quote (same as `literal_quote`).
    - Use within default implementation of the `escape_single_quotes` macro.

These low-level configurations would then inform the intermediate-level macros like the following:
- `string_literal`
- `escape_single_quotes` (maybe rename this to something else more abstract)

## References
1. https://www.ibm.com/docs/en/informix-servers/12.10?topic=statements-quotation-marks-escape-characters
2. https://4js.com/online_documentation/fjs-fgl-3.00.05-manual-html/c_fgl_sql_programming_080.html

Reply:
## Sidenote
Some macros assume single quotes to demarcate string literals rather than utilizing `string_literal`. [This](https://github.com/calogica/dbt-expectations/blob/73c8ab3b6e313c5b95c33abde329de682d51bb7f/macros/schema_tests/distributional/expect_row_values_to_have_data_for_every_n_datepart.sql#L60) is one example. Not sure how common this is or not.

Reply:
Done.